modules/custom/az_migration/src/Plugin/migrate/process/MediaBundleRecognizer.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Process plugin to recognize which media bundle to use.
+ *
+ * Available configuration keys
+ * - prefix: The prefix to use when determining the entity type.
+ *
+ * Examples:
+ *
+ * Consider a paragraphs migration, where you want to be able to automatically
+ * use a specific destination media type but only use one media migration.
+ *
+ * @code
+ * process:
+ *   destination_bundle:
+ *     plugin:  az_media_bundle_recognizer
+ *     prefix: 'az_'
+ *
+ * @endcode
+ *
+ * @MigrateProcessPlugin(
+ *   id = "az_media_bundle_recognizer"
+ * )
+ *
+ * @ return String that represents a bundle machine name.
+ */
+class MediaBundleRecognizer extends ProcessPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+
+    return $instance;
+  }
+
```

Comment:
I don't think this `create` method (or the `ContainerFactoryPluginInterface`) are needed since you're not adding any additional dependencies.
```suggestion
```